### PR TITLE
Better readme

### LINKED ISSUE #2: Update README
Our readme is pretty anemic right now. Once we have our initial setup finished, it should be updated to reflect all the information at the time.

We should add more information about the functionality available as well.

### PR DESCRIPTION
**Pull request by skeet70**
This should close #2. 

Adds [tut](https://github.com/tpolecat/tut) to give us real REPL output in our README and build time checking of our documentation statements.

Also adds real documentation with examples for all the syntax that we're currently adding with the library.

### CODE REVIEW

**Comment by skeet70** [comment]
You probably want to view https://github.com/skeet70/rubiz/blob/584b4622ecc2c84c8614cdc4f42730f9cf62f1bc/README.md instead of the raw .md to see most of the changes.

File: README.md [hunk 1]
```diff
@@ -4,29 +4,232 @@ Rubiz
...
+```scala
+(Task.now(List("Australia", "Japan"))
+  .withTiming           // Task[(FiniteDuration, List[String])]
+  .map {
+      case (timing, result) =>
+        println(s"${result.length} country names were returned by the DB in ${timing.toMillis} ms.")
+        result
+  }
+  .run)
+// 2 country names were returned by the DB in 2 ms.
```
Comment (coltfred) [inline README.md]: There isn't anything about a DB here.

File: README.md [hunk 2]
```diff
@@ -4,29 +4,232 @@ Rubiz
...
+failed, there isn't a result to wrap up with, and no timing information
+will be available. This is most useful for local logging of timing
+information.
+
+```scala
+(Task.now(List("Australia", "Japan"))
+  .withTiming           // Task[(FiniteDuration, List[String])]
+  .map {
+      case (timing, result) =>
+        println(s"${result.length} country names were returned by the DB in ${timing.toMillis} ms.")
```
Comment (coltfred) [inline README.md]: There isn't a DB.

File: README.md (hunk 1 above)
Comment (skeet70) [reply README.md]: It's an example, a bit more useful to consider timing on things like that than a literal list. Can you think of a better way to exemplify that without an actual DB, or make it clear that the print statement is an example?

**Review by coltfred — COMMENTED** [review]
A couple small changes.

**Review by leifwickland — APPROVED** [review]
Awesome work!

File: README.md (hunk 1 above)
Comment (skeet70) [reply README.md]: @coltfred ?

Comment (coltfred) [reply README.md]: You could sleep in a `Task.delay` if you wanted to show something.

**Pull request merged**